### DISCUSSION TITLE
How can I commit and push the commit under another user's name?

### DISCUSSION THREAD

**Discussion by scriptxyz** [gitpython-developers/GitPython#1257] [Q&A] [answered]
Hi,

I'm trying to write a script and running the script on a linux box. I need to use an user account to connect to the git repo

Currently I'm using below command to clone the repo:
```
remote_url = "ssh://UserA@git.corp.com:/package"
git_env = {"GIT_SSH_COMMAND":"ssh -i key -o StricktHostKeyChecking=no“”｝
Repo.clond_from(remote_url, ”/tmp/repo",env=git_env}

```

The clone command works fine .But I need to commit changes and push the changes under this user account.
I didn't find any document about how to set env when commit and push the change.

Any suggestion?

**Comment by Byron** [comment, marked as answer]
In the [documentation about remotes](https://gitpython.readthedocs.io/en/stable/tutorial.html#handling-remotes) you will find this:

```python
ssh_cmd = 'ssh -i id_deployment_key'
with repo.git.custom_environment(GIT_SSH_COMMAND=ssh_cmd):
    repo.remotes.origin.fetch()
```

which applies to pushes as well.
